cli/util/util.go

```diff
@@ -26,3 +31,39 @@ func Dial(addr, certPEM string) (*grpc.ClientConn, error) {
 		grpc.WithDefaultCallOptions(grpc.UseCompressor(gzip.Name)),
 		grpc.WithUnaryInterceptor(errors.UnaryClientInterceptor))
 }
+
+func GetDockerClient() (*client.Client, error) {
+	dockerClient, dcErr := client.NewClientWithOpts(client.FromEnv, client.WithAPIVersionNegotiation())
+
+	// If a custom host is set, don't try any funny business.
+	if dcErr != nil || dockerClient.DaemonHost() != client.DefaultDockerHost {
+		return dockerClient, dcErr
+	}
+
+	socketUrl, err := url.Parse(dockerClient.DaemonHost())
+	if err != nil {
+		// We really do not expect this to ever happen.
```

Comment:
I'm fine with this panic, but I generally prefer to just return errors unless there's a clear benefit to panicking. We should agree on a style at some point.

Reply:
Here, I felt a panic is appropriate because `dockerClient.DaemonHost() == client.DefaultDockerHost`, so failing to parse would mean that `client.DefaultDockerHost` is malformed, which should be impossible to trigger. IMO, that should be the style guideline: panic only in scenarios that should be literally impossible unless the code is not working correctly.

Reply:
I could use `client.DefaultDockerHost` on line 43 if that would make things clearer.

Reply:
It's fine as is. Let's just see if it comes up again in the future. I think we both agree that it'd be bad to panic when it's a user error rather than a code error. So I just don't want us to have unnecessary mental overhead when writing/reading code to make sure that an error couldn't also be caused by a user action.